weaviate/connect/connection.py

```diff
@@ -272,6 +281,7 @@ def __init__(
             self._headers["authorization"] = "Bearer " + auth_client_secret.api_key
 
         self._session: Session
+        self._httpx_async_client: httpx.AsyncClient = httpx.AsyncClient()
```

Comment:
The created client here needs to be closed at some point in order that the connections are automatically cleaned-up using `await client.aclose()`

When we add `async` fully, I think it would make sense to either have a `def close()` method in `WeaviateClient` that users must call themselves to clean up connections or we make the `WeaviateClient` a context manager where the `__exit__` method cleans up hanging connections

Unfortunately, the `__del__` method cannot guarantee that the connections are closed properly :(

However, for now, I think we can avoid all this by having the `httpx.AsyncClient` created and closed within the same context as the batching client itself through `_BatchWrapper`, since it is only used in that context anyway

Would that be a large refactor?